### ISSUE TITLE
Factorie crashes on some documents

### ISSUE THREAD

**Issue by nopper** [factorie/factorie#335]
I've been using factorie to annotate various documents and I am experiencing various crashes here and there. I am using the following parameters:

    $ bin/fac nlp --token --wsj-forward-pos --transition-based-parser --conll-chain-ner

For example annotating the following document:

```
$ nc localhost 3228
BASEBALL- MAJOR LEAGUE STANDINGS AFTER THURSDAY'S GAMES. NEW YORK 1996-08-23 Major League Baseball standings after games played on Thursday( tabulate under won, lost, winning percentage and games behind): AMERICAN LEAGUE EASTERN DIVISION W L PCT GB NEW YORK 72 54.571- BALTIMORE 67 59.532 5 BOSTON 64 64.500 9 TORONTO 59 69.461 14 DETROIT 45 82.354 27 1/2 CENTRAL DIVISION CLEVELAND 76 51.598- CHICAGO 69 60.535 8 MINNESOTA 63 64.496 13 MILWAUKEE 60 68.469 16 1/2 KANSAS CITY 58 71.450 19 WESTERN DIVISION TEXAS 74 54.578- SEATTLE 65 61.516 8 OAKLAND 62 68.477 13 CALIFORNIA 59 68.465 14 1/2 FRIDAY, AUGUST 23 SCHEDULE SEATTLE AT BOSTON MILWAUKEE AT CLEVELAND CALIFORNIA AT BALTIMORE OAKLAND AT NEW YORK TORONTO AT CHICAGO DETROIT AT KANSAS CITY TEXAS AT MINNESOTA NATIONAL LEAGUE EASTERN DIVISION W L PCT GB ATLANTA 79 47.627- MONTREAL 68 58.540 11 NEW YORK 59 69.461 21 FLORIDA 58 69.457 21 1/2 PHILADELPHIA 52 76.406 28 CENTRAL DIVISION HOUSTON 68 60.531- ST LOUIS 67 60.528 1/2 CHICAGO 63 62.504 3 1/2 CINCINNATI 63 62.504 3 1/2 PITTSBURGH 54 73.425 13 1/2 WESTERN DIVISION SAN DIEGO 70 59.543- LOS ANGELES 67 60.528 2 COLORADO 66 62.516 3 1/2 SAN FRANCISCO 54 71.432 14 FRIDAY, AUGUST 23 SCHEDULE CINCINNATI AT FLORIDA( doubleheader) CHICAGO AT ATLANTA ST LOUIS AT HOUSTON PITTSBURGH AT COLORADO NEW YORK AT LOS ANGELES PHILADELPHIA AT SAN DIEGO MONTREAL AT SAN FRANCISCO
```

Produces the following stack trace:

```
Exception in thread "ServerThread" java.lang.Error: Invalid combination of states, prefix L, state NotReading at token Token(1325:PHILADELPHIA)
        at cc.factorie.app.nlp.ner.NerAnnotator$$anonfun$process$1.apply(NerAnnotator.scala:53)
        at cc.factorie.app.nlp.ner.NerAnnotator$$anonfun$process$1.apply(NerAnnotator.scala:28)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
        at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at cc.factorie.app.nlp.ner.NerAnnotator.process(NerAnnotator.scala:28)
        at cc.factorie.app.nlp.DocumentAnnotationPipeline$$anonfun$process$2.apply(DocumentAnnotatorPipeline.scala:37)
        at cc.factorie.app.nlp.DocumentAnnotationPipeline$$anonfun$process$2.apply(DocumentAnnotatorPipeline.scala:35)
        at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:778)
        at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:777)
        at cc.factorie.app.nlp.DocumentAnnotationPipeline.process(DocumentAnnotatorPipeline.scala:35)
        at cc.factorie.app.nlp.NLP$ServerThread$$anonfun$run$1.apply(NLP.scala:102)
        at cc.factorie.app.nlp.NLP$ServerThread$$anonfun$run$1.apply(NLP.scala:100)
        at scala.collection.Iterator$class.foreach(Iterator.scala:742)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1194)
        at cc.factorie.app.nlp.NLP$ServerThread.run(NLP.scala:100)
```

**Comment by strubell** [comment]
Can you post a minimal example?

**Comment by nopper** [comment]
Yep sorry I am trying to understand which document in my pipeline triggers the exception. I accidentally hit the New Issue button ;) I'll update the bug report in a few minutes

**Comment by Craigacp** [comment]
That's from the NER changes. If you update to the latest jar it should turn that message into a logger warning rather than an Error (as you have NerAnnotator rather than NerChunkAnnotator).

**Comment by nopper** [comment]
Ok I've update (`git pull`) and recompiled everything with:

    $ ./sbt -J-Xmx4G with-nlp-resources:assembly

But now I am unable to run the server:

```
$ bin/fac nlp --token --wsj-forward-pos --transition-based-parser --conll-chain-ner
java -Xmx6g -ea -Djava.awt.headless=true -Dfile.encoding=UTF-8 -server -classpath ./src/main/resources:./target/classes:./target/factorie-1.2-SNAPSHOT-nlp-jar-with-dependencies.jar
Exception in thread "main" java.lang.Error: No file named Nothing$.factorie found in classpath for class scala.runtime.Nothing$, and no value found in system property scala.runtime.Nothing$. To fix this either add a file with the right name to the classpath or set the system property to point to a directory containing the file.
        at cc.factorie.util.ClasspathURL$.apply(ClasspathURL.scala:42)
        at cc.factorie.util.ModelProvider$$anon$1.<init>(ModelProvider.scala:81)
        at cc.factorie.util.ModelProvider$.classpath(ModelProvider.scala:80)
        at cc.factorie.app.nlp.ner.ConllChainNer$.<init>(ChainNer.scala:58)
        at cc.factorie.app.nlp.ner.ConllChainNer$.<clinit>(ChainNer.scala)
        at cc.factorie.app.nlp.NLP$opts$2$$anon$8.invoke(NLP.scala:49)
        at cc.factorie.util.CmdOptions$CmdOption.parse(CmdOption.scala:174)
        at cc.factorie.util.CmdOptions.parse(CmdOption.scala:111)
        at cc.factorie.app.nlp.NLP$.main(NLP.scala:66)
        at cc.factorie.app.nlp.NLP.main(NLP.scala)
```

**Comment by strubell** [comment]
Two things:
1) It looks like we have no serialized model for conll-chain-ner in our package of serialized models! I will train and deploy one today, and comment on this issue. In the meantime, using --stacked-chain-ner instead of --conll-chain-ner should work. 

2) I made a lot of changes to the parser which required deploying a new serialized model, but I just realized that I didn't change our build scripts to depend on the new all-models snapshot that contains the new model; I am pushing those changes now, so please pull and build again along with the ner model change

Sorry for the difficulty, we've been making a lot of changes to our nlp pipeline in the past few days!

**Comment by strubell** [comment]
Oops, ok, looks like stacked-chain-ner isn't working either, looking into this

**Comment by JackSullivan** [comment]
With c0921fab15c9e3920127376d563f5ccf7faca6fa, this issues should be fixed. Please give it another shot and let me know if you are still getting an error.

**Comment by nopper** [comment]
Ok I've pulled the changes. Now the `bin/fac` script seems to be broken:

```
$ bin/fac nlp --token --wsj-forward-pos --transition-based-parser --stacked-chain-ner
./target/factorie_2.11-1.2-SNAPSHOT-jar-with-dependencies.jar not found
```

I've edited in order to make it run. But now it does not accept the `--stacked-chain-ner` for some reason.

```
$ bin/fac nlp --token --wsj-forward-pos --transition-based-parser --stacked-chain-ner
java -Xmx6g -ea -Djava.awt.headless=true -Dfile.encoding=UTF-8 -server -classpath ./src/main/resources:./target/classes:./target/factorie-1.2-SNAPSHOT-nlp-jar-with-dependencies.jar
Unrecognized option --stacked-chain-ner
Usage: --ontonotes-chain-ner=URL --socket=SOCKETNUM --config=FILE --log=FILENAME --token= --ontonotes-forward-pos=URL --parse-forward-coref=URL --ontonotes-chain-pos=URL --wsj-forward-pos=URL --encoding=ENCODING --transition-based-parser=URL --wordnet-lemma=URL --parse-lexical-coref=URL --conll-chain-ner=URL --stacked-chain-ner-noembeddings=URL --sentence= --ner-forward-coref=URL --help=STRING --d-named-coref= --version=STRING --ner-lexical-coref=URL --token-norm=
```

Removing the command line flag seems to make it running again.

**Comment by JackSullivan** [comment]
I'll look in to stacked-chain-ner, can you post the diff that you made to `bin/fac`, honestly I've never used the script before.

**Comment by strubell** [comment]
My mistake, it's actually --stacked-chain-ner-noembeddings, and in fact --conll-chain-ner should also work after all

**Comment by strubell** [comment]
@JackSullivan this is still unresolved

**Comment by strubell** [comment]
Ok, this should be fixed in PR #339 let us know if you are still having issues. I'm now working on an additional fix that will resolve the incorrect paths in the bin/fac script (sbt and maven are currently creating jars with different names, looks like it should be straightforward to fix).

**Comment by strubell** [comment]
Ok, script issues should be resolved by PR #340
